components/support/ffi/src/macros.rs

```diff
@@ -195,3 +195,15 @@ pub fn convert_to_json_string<T: serde::Serialize>(value: &T) -> *mut c_char {
     let as_string = serde_json::to_string(&value).unwrap();
     rust_string_to_c(as_string)
 }
+
+/// Define a (public) destructor the ByteBuffer type. It is common to define
```

Comment:
You should probably use `define_string_destructor!` as a model for this comment. This is somewhat confusing and it's much better to be explicit about these things (and even in this patch, you get it subtly wrong in fxa-client's ffi.rs, although in a way that only matters for debug builds, and that will fail loudly instead of corrupting memory or something).

Also, I'd prefer if you moved this so it was near the other `define_foo_destructor!` macros.